src/features/toilet/Toilet.js

```diff
@@ -1,22 +1,239 @@
+import ButtonFluid from "../../common/components/buttons/ButtonFluid";
 import HeaderSub from "../../common/components/headers/HeaderSub";
+import ListDefault from "../../common/components/lists/ListDefault";
+import ReviewCard from "../../common/components/reviewCard/ReviewCard";
+import StarContainer from "../../common/components/starContainer/StarContainer";
+import Title from "../../common/components/Title";
+import { updateToiletInfo } from "./toiletSlice";
 
 const StyledToilet = styled.div`
   width: 100%;
   display: flex;
   flex-direction: column;
   background-color: black;
+  color: white;
+
+  .titleContainer {
+    display: flex;
+    align-items: center;
+  }
+
+  .rankContainer {
+    display: flex;
+    align-items: center;
+    justify-content: space-evenly;
+  }
+
+  .toiletPaperContainer {
+    width: 100%;
+    display: flex;
+    align-items: center;
+  }
+
+  .lastToiletPaterProvideTime {
+    width: 100%;
+    font-size: small;
+    color: gray;
+    margin-right: 1rem;
+  }
 `;
 
 function Toilet() {
-  // const navigate = useNavigate();
+  const { toilet_id } = useParams();
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const location = useLocation();
+
+  const {
+    showToiletPath,
+    toiletName,
+    roadNameAddress,
+    inUnisexToilet,
+    menToiletBowlNumber,
+    menHandicapToiletBowlNumber,
+    menChildrenToiletBowlNumber,
+    ladiesToiletBowlNumber,
+    ladiesHandicapToiletBowlNumber,
+    ladiesChildrenToiletBowlNumber,
+    openTime,
+    latestToiletPaperInfo,
+    isSOS,
+    chatRoomList,
+  } = location.state;
+
+  const userClickedSOSButton = useSelector(
+    (state) => state.currnetToiletInfo.byIds[toilet_id]?.userClickedSOSButton
+  );
+  const isSOSCurrnetToilet = useSelector(
+    (state) => state.currnetToiletInfo.byIds[toilet_id]?.isSOSCurrnetToilet
+  );
+
+  const [reviews, setReviews] = useState([]);
+  const [avgRating, setAvgRating] = useState(0);
+  const [userSOSButton, setUserSOSButton] = useState(userClickedSOSButton);
+
+  useEffect(() => {
+    dispatch(updateToiletInfo({ toilet_id, isSOS, userSOSButton }));
+  }, [userSOSButton]);
+
+  useEffect(() => {
+    async function getReviews() {
+      try {
+        const { data } = await axios.get(`/toilets/review/${toilet_id}`);
+        setReviews(data.reviewList);
+      } catch (error) {
+        // 추후 에러처리 필요.
+        console.log(error);
+      }
+    }
+
+    getReviews();
+  }, []);
+
+  useEffect(() => {
+    function getAvgRating() {
+      if (!reviews.length) {
+        return 0;
+      }
+
+      let totalRating = 0;
+      // eslint-disable-next-line no-return-assign
+      reviews.forEach((review) => (totalRating += review.rating));
+      return totalRating / reviews.length;
+    }
+    setAvgRating(getAvgRating());
+  }, [reviews]);
+
+  function onClickSOSButton() {
+    setUserSOSButton(true);
+
+    async function occurSOS() {
```

Comment:
제 개인적으로 느끼기에 occur 라는 단어는 수동의 의미가 강한것 같은데요. 여기서는 유저가 SOS 버튼을 능동적으로 클릭한걸 처리하는 거니까 emitSOS 등의 네이밍은 어떨까요?
```suggestion
    async function emitSOS() {
```

Reply:
emitSOS 가 더 좋은것 같아요. 이 부분은 server 부분까지 수정해서 다시 PUSH 하겠습니다.